demo/src/features/egui/jobs/extract.rs

```diff
@@ -77,6 +82,8 @@ impl<'extract> ExtractJobEntryPoints<'extract> for EguiExtractJob<'extract> {
                 egui_draw_data,
                 egui_material_pass: self
                     .asset_manager
+                    .get()
```

Comment:
All of the calls to `get().unwrap()` seem the opposite of ergonomic.

Reply:
Would it be possible to still implement `Deref` and have it automatically do the equivalent of `get().unwrap()`?

Reply:
I don't like it, but I also don't like a Deref that panics. I didn't use just unwrap() because unwrap() generally consumes self. Do you think a different name would be better? Like unwrap_ref(), get_or_panic(), or something like that?

Reply:
What if we had `fetch` return a `ReadBorrow` like before, and added `try_fetch` that would return an `Option<ReadBorrow>` -- does that solve the original problem still? I think that seems more ergonomic since if a resource is required to be present one could use `fetch` without this `get().unwrap()` boilerplate, whereas if it is only conditionally present you use `try_fetch` and bind the `Option` accordingly.

Reply:
Found a different approach that lets us take a "borrow" when creating the extract job.